### PR TITLE
bug fix: zero the grads on the patch after each iteration

### PR DESCRIPTION
**Pull request by mbc**
# Description

In the `AdversarialTexturePyTorch` attack, the gradients to the patch were not being zero'd after iteration; thus, they were be accumulated as the attack went on.

## Type of change

- [ ] Improvement (non-breaking)
- [X] Bug fix (non-breaking)
- [ ] New feature (non-breaking)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Testing

- Tested attacks on a few samples by hand.
- Used a modified version of the attack that accepts an initial patch as input; confirmed that, e.g. running the attack twice, sequentially, for N iterations each time, is equivalent to running it once for 2N iterations (which was *not* the case previously).

# Checklist

- [ ] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] I have commented my code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: art/attacks/evasion/adversarial_texture/adversarial_texture_pytorch.py [hunk 1]
```diff
@@ -170,6 +170,8 @@ def _train_step(
                 y=None,
             )
 
+        self._patch.grad = torch.zeros(self._patch.grad.shape, device=self._patch.grad.device, dtype=self._patch.grad.dtype)
+
```
Comment (beat-buesser) [inline art/attacks/evasion/adversarial_texture/adversarial_texture_pytorch.py:174]:
```suggestion
        self._patch.grad = torch.zeros(
            self._patch.grad.shape, device=self._patch.grad.device, dtype=self._patch.grad.dtype
        )

```

**Review by beat-buesser — APPROVED** [review]
Hi @mbc Thank you very much for reporting this bug and the solution with your pull request! I have added one commit to fix the formatting.

**Comment by beat-buesser** [comment]
I'll pull this PR to branch `bugfix-carla-objtrack-zero-grads` to be able to add a commit for the formatting updates and to release it with ART 1.10.3 today.
